zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -771,8 +771,12 @@ object Http {
   }
 
   final case class PartialCollect[A](unit: Unit) extends AnyVal {
-    def apply[B](pf: PartialFunction[A, B]): Http[Any, Nothing, A, B] =
-      FromFunctionHExit(a => if (pf.isDefinedAt(a)) HExit.succeed(pf(a)) else HExit.Empty)
+    def apply[B](pf: PartialFunction[A, B]): Http[Any, Nothing, A, B] = {
```

Comment:
run the benchmarks on CI on main and current branch so that we can see the difference